### PR TITLE
[REVIEW] PyArrow: Fix issue in typing of scalars, handle date32 and date64 types

### PR DESCRIPTION
**Pull request by kkraus14**

### CODE REVIEW

**Comment by scopatz** [comment]
Seems to fail with 

```
12:58:48         assert isinstance(pa_gdf, pa.Table)
12:58:48 >       assert pa.Table.equals(pa_df, pa_gdf)
12:58:48 E       AssertionError: assert False
12:58:48 E        +  where False = <method 'equals' of 'pyarrow.lib.Table' objects>(pyarrow.Table\na: timestamp[ms]\nb: timestamp[ms], pyarrow.Table\na: date64[ms]\nb: date64[ms])
12:58:48 E        +    where <method 'equals' of 'pyarrow.lib.Table' objects> = <class 'pyarrow.lib.Table'>.equals
12:58:48 E        +      where <class 'pyarrow.lib.Table'> = pa.Table
12:58:48 
12:58:48 pygdf/tests/test_dataframe.py:750: AssertionError
```

**Comment by kkraus14** [comment]
@scopatz issue is that we use datetime64[ms] and when pushing to an arrow table should create pa.date64 columns, whereas Pandas converts its datetime64[ns] to pa.timestamp('ns'). Added some nasty typecasting in pyarrow (timestamp('ns') --> timestamp('ms') --> int64 --> date64) to match test output properly.

File: pygdf/columnops.py [hunk 1]
```diff
@@ -197,6 +197,24 @@ def as_column(arbitrary):
...
+                np.dtype('M8[ms]')
+            ))
+            data = datetime.DatetimeColumn(
+                data=padata,
+                mask=pamask,
+                null_count=arbitrary.null_count,
+                dtype=np.dtype('M8[ms]')
+            )
+        elif isinstance(arbitrary, pa.Date32Array):
+            # No equivalent np dtype and not yet supported
```
Comment (dantegd) [inline pygdf/columnops.py]: wouldn't it be good to throw a warning for users?

**Comment by dantegd** [comment]
looks good, merging

**Pull request merged**